### PR TITLE
Support test and filter plugins in ansible-doc sanity test

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
For these plugin types, simply list all plugins, and pass all names that don't have `UNDOCUMENTED` next to them to `ansible-doc`.

This should also not be affected by planned refactorings for sidecar docs.

CC @bcoca

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ansible-doc sanity test

### CODE REVIEW

**Comment by felixfontein** [comment]
Interestingly this fails in community.general as for some reason, `ansible-doc -l -t filter` does find two documented filters `community.general.lists_mergeby` and `community.general.groupby_as_dict` that `ansible-doc --metadata-dump -t filter` and `ansible-doc -t filter` does *not* find:
```.bash
> ansible-doc -t filter community.general.lists_mergeby community.general.groupby_as_dict
[WARNING]: filter community.general.lists_mergeby not found in: /home/me/.ansible/plugins/filter:/usr/share/ansible/plugins/filter:/path/to/ansible/lib/ansible/plugins/filter
[WARNING]: filter community.general.groupby_as_dict not found in: /home/me/.ansible/plugins/filter:/usr/share/ansible/plugins/filter:/path/to/ansible/lib/ansible/plugins/filter
```

**Comment by felixfontein** [comment]
@bcoca is this possibly because the docs are contained in the `.py` file, but the `.py` file is not named the same as the filter?

**Comment by felixfontein** [comment]
Needs #78700 to work with collections, and a fix for #78786 to work for ansible-core.

**Comment by felixfontein** [comment]
Now includes #78788.

**Comment by felixfontein** [comment]
No longer includes #78788 (that has been merged).

**Comment by bcoca** [comment]
most filters are waiting for my PR to get docs, how is this passing?

**Comment by felixfontein** [comment]
@bcoca this test does not ensure that all filters and tests have docs, but that the docs - when present - don't make `ansible-doc -t <type> <plugin_name>` crash (because they are broken, invalid, ...). Basically the trick is to ignore all tests/filters whose description returned by listing is `UNDOCUMENTED`: https://github.com/ansible/ansible/pull/77737/files#diff-1d2ebb1b245cdbbb2eacce44957ba100eadd6d71dc24a672bae55410c65326c2R118

(I would also argue that a test that *enforces* that all filters/tests have docs isn't a good idea, without giving collections at least some time to adapt this feature first.)

File: changelogs/fragments/77737-ansible-doc-test-filters-tests.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "ansible-doc sanity test - now also lists documentation for test and filter plugins that are documented (https://github.com/ansible/ansible/pull/77737)."
```
Comment (s-hertel) [inline changelogs/fragments/77737-ansible-doc-test-filters-tests.yml]:
```suggestion
  - "ansible-test sanity --test ansible-doc - now also lists documentation for test and filter plugins that are documented (https://github.com/ansible/ansible/pull/77737)."
```

**Review by s-hertel — APPROVED** [review]
Not a blocker, but I'm not sure why `plugins/filter/check_pylint.py` is still showing up in the ansible-test-sanity results. Does that need to be renamed somewhere else too?

```
Run command with data: importer.py
...
Command exited with status 10 after 0.2035222053527832 seconds.
plugin: plugins/filter/check_pylint.py
plugins/lookup/bad.py
plugins/lookup/world.py
plugins/plugin_utils/check_pylint.py
```

**Comment by felixfontein** [comment]
> Not a blocker, but I'm not sure why `plugins/filter/check_pylint.py` is still showing up in the ansible-test-sanity results. Does that need to be renamed somewhere else too?

My guess is that it only shows up in this PR since the file was moved, and change detection triggers tests both for the old and new location. It should go away once this is merged...

**Comment by s-hertel** [comment]
@felixfontein Oh, maybe so. I was running the tests locally, I didn't think that took into account change detection. :confused:

**Comment by felixfontein** [comment]
@s-hertel thanks for reviewing and merging this!

**Pull request merged**